src/main/java/net/minecrell/serverlistplus/core/status/PersonalizedStatusPatch.java

```diff
@@ -70,63 +72,146 @@ public boolean hasChanges() {
     public String getPlayerSlots(StatusResponse response) {
         List<String> result;
-        return prepareRandomEntry(response,
-                response.getRequest().isIdentified() && (result = personalized.getSlots()) != null ?
-                        result : def.getSlots());
+        if (response.getRequest().isIdentified()) {
+            if (response.getRequest().getIdentity().isBanned(response.getCore())
+                    && banned.getSlots() != null) {
+                return prepareRandomEntry(response, banned.getSlots());
+            }
+
+            if (personalized.getSlots() != null) {
+                return prepareRandomEntry(response, personalized.getSlots());
+            }
+        }
+
+        return prepareRandomEntry(response, def.getSlots());
     }
 
     public String getVersion(StatusResponse response) {
         List<String> result;
-        return prepareRandomEntry(response,
-                response.getRequest().isIdentified() && (result = personalized.getVersions()) != null ?
-                        result : def.getVersions());
+        if (response.getRequest().isIdentified()) {
+            if (response.getRequest().getIdentity().isBanned(response.getCore())
+                    && banned.getVersions() != null) {
+                return prepareRandomEntry(response, banned.getVersions());
+            }
+
+            if (personalized.getVersions() != null) {
+                return prepareRandomEntry(response, personalized.getVersions());
+            }
+        }
+
+        return prepareRandomEntry(response, def.getVersions());
     }
 
     public Integer getProtocolVersion(StatusResponse response) {
         Integer result;
-        return response.getRequest().isIdentified() && (result = personalized.getProtocolVersion()) != null ?
-                result : def.getProtocolVersion();
+        if (response.getRequest().isIdentified()) {
+            if (response.getRequest().getIdentity().isBanned(response.getCore())
+                    && banned.getProtocolVersion() != null) {
+                return banned.getProtocolVersion();
+            }
+
+            if (personalized.getProtocolVersion() != null) {
+                return personalized.getProtocolVersion();
+            }
+        }
+
+        return nextNumber(nextEntry(def.getMax()));
     }
 
     public FaviconSource getFavicon(StatusResponse response) {
         List<FaviconSource> result;
-        FaviconSource favicon = nextEntry(
-                response.getRequest().isIdentified() && (result = personalized.getFavicons()) != null ?
-                        result : def.getFavicons());
+        FaviconSource favicon = null;
+        if (response.getRequest().isIdentified()) {
+            if (response.getRequest().getIdentity().isBanned(response.getCore())
+                    && banned.getFavicons() != null) {
+                favicon = nextEntry(banned.getFavicons());
+            } else if (personalized.getFavicons() != null) {
```

Comment:
No, else would be wrong because it should still apply the personalized favicon if present even if the banned favicon is **not** present.